nixos/modules/services/misc/rkvm.nix

```diff
@@ -0,0 +1,168 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
```

Comment:
Why? ripgrep in Doom shows 23139 usages of `with lib;` in nixpkgs:<br>
![screenshot](https://github.com/NixOS/nixpkgs/assets/25263210/d88bb6e4-765b-44fa-b941-13902ae175c8)

Reply:
I believe there are ongoing efforts to remove most top-level `with lib;` statements, due to some language intricacies. There's also this page: https://nix.dev/recipes/best-practices#with-scopes, which uses `pkgs` as an example but lists out some of the concerns.

I think the most maintainable solution is to use an `inherit (lib) ...` at the top for the most used stuff (like `mkOption` and `types`), and just prepend `lib.` for every function you only use once or twice. That way, there's no need to maintain a big list of everything used up at the top, while still keeping most of the code relatively clean.

Output of `git log --grep "with lib"` shows some of the efforts:

![image](https://github.com/NixOS/nixpkgs/assets/14929991/fb0a1dd9-8f47-4470-aa87-fb99c1b83ef0)

Reply:
I'm not going to block this PR on this issue, but I'll link #208242. I don't really mind either outcome, but it would be good to come to some sort of agreement, so people don't have to go back and forth on this for every PR.